snap/sysparams/sysparams_test.go

```diff
@@ -0,0 +1,152 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2023 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package sysparams_test
+
+import (
+	"errors"
+	"io/ioutil"
+	"os"
+	"path"
+	"testing"
+
+	. "gopkg.in/check.v1"
+
+	"github.com/snapcore/snapd/dirs"
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/snap/sysparams"
+	"github.com/snapcore/snapd/testutil"
+)
+
+func TestSysParamsTest(t *testing.T) { TestingT(t) }
+
+type sysParamsTestSuite struct {
+	testutil.BaseTest
+}
+
+var _ = Suite(&sysParamsTestSuite{})
+
+func (s *sysParamsTestSuite) SetUpTest(c *C) {
+	dirs.SetRootDir(c.MkDir())
+}
+
+func (s *sysParamsTestSuite) TestOpenNewEmpty(c *C) {
+	// Opening the file when it doesn't exist, should not create
+	// the file unless Update is called. And that an empty update
+	// provides empty members
+	sspPath := dirs.SnapSystemParamsUnder(dirs.GlobalRootDir)
+	ssp, err := sysparams.Open(sspPath)
+	c.Check(err, IsNil)
+	c.Assert(ssp, NotNil)
+	c.Check(sspPath, testutil.FileAbsent)
+
+	r := sysparams.MockOsutilAtomicWriteFile(func(filename string, data []byte, perm os.FileMode, flags osutil.AtomicWriteFlags) error {
+		c.Check(string(data), Equals, "homedirs=\n")
+		return nil
+	})
+	defer r()
+
+	// Save the file
+	err = ssp.Write()
```

Comment:
this test one can probably not mock osutil.AtomicWriteFile and check the file on disk and its permissions